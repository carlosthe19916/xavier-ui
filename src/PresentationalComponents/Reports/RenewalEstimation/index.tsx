import RenewalEstimation from './RenewalEstimation';

export default RenewalEstimation;
