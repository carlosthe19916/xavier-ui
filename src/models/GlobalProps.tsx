export interface GlobalProps {
    history: any;
    location: any;
    match: any;
}
