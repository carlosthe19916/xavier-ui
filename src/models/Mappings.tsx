export interface FlagAssessmentModel {
    flag: string;
    label: string;
    flagAssessmentOSModels: FlagAssessmentOSModel[]
}

export interface FlagAssessmentOSModel {
    osName: string | undefined;
    assessment: string;
}
