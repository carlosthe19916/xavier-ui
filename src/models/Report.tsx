export interface Report {
    id: number;
    customerId: string;
    fileName: string;
    numberOfHosts: number;
    totalDiskSpace: number;
    totalPrice: number;
    creationDate: number
}
