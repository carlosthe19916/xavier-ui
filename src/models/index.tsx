export * from './Report';
export * from './Upload';
export * from './User';
export * from './Mappings';
