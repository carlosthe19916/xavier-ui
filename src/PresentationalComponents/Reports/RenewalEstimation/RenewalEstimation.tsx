import React, { Component } from 'react';
import {
    Table,
    TableBody,
    TableVariant,
    TableHeader
} from '@patternfly/react-table';
import { formatValue } from '../../../Utilities/formatValue';
import { SourceCostsModel } from '../../../models';

interface Props {
    data: SourceCostsModel
}

interface State {
}

class RenewalEstimation extends Component<Props, State> {

    constructor(props: Props) {
        super(props);
    }

    render() {
        const { data: {
            sourceRenewHighValue,
            sourceRenewLikelyValue,
            sourceRenewLowValue
        }} = this.props;

        const columns = [
            { title: 'For vSphere Enterprise Plus' },
            { title: 'ELA Renewal Estimation', props: { className: 'pf-u-text-align-right' }}
        ];

        const rows = [
            [
                '-- As high as...',
                formatValue(sourceRenewHighValue, 'usd', { fractionDigits: 0 })
            ],
            [
                '-- Most likely...',
                formatValue(sourceRenewLikelyValue, 'usd', { fractionDigits: 0 })
            ],
            [
                '-- As low as...',
                formatValue(sourceRenewLowValue, 'usd', { fractionDigits: 0 })
            ]
        ];

        return (
            <React.Fragment>
                <Table aria-label='Environment'
                    cells={ columns }
                    rows={ rows }
                    variant={ TableVariant.compact }
                    borders={ false }
                >
                    <TableHeader />
                    <TableBody />
                </Table>
            </React.Fragment>
        );
    }
}

export default RenewalEstimation;
