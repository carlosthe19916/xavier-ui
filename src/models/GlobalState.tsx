import { Report } from './Report';
import { Upload } from './Upload';

export interface GlobalState {
    reports: {
        report: Report,
        reports: Report[];
        loading: boolean;
        error: string;
        total: number;
    },
    uploads: {
        uploads: Upload[]
    }
}
