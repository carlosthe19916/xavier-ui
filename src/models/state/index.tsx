import { Report } from '../Report';
import { Upload } from '../Upload';

export interface GlobalState {
    reports: ReportState,
    uploads: UploadState
}

export interface ReportState {
    error: string | null;
    total: number;
    report: Report | null;
    reports: Report[];
    loading: boolean;
}

export interface UploadState {
    uploads: Upload[]
}