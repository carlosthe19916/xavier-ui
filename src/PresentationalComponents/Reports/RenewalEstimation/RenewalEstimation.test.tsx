import { shallow } from "enzyme";
import RenewalEstimation from "./RenewalEstimation";
import { SourceCostsModel } from "../../../models";

const baseSourceCostsModel: SourceCostsModel = {
  sourceRenewHighValue: 1000,
  sourceRenewLikelyValue: 2000,
  sourceRenewLowValue: 3000
};

describe("RenewalEstimation", () => {
  it("expect to render", () => {
    const data: SourceCostsModel = {
      ...baseSourceCostsModel
    };

    const wrapper = shallow(
      <RenewalEstimation
        data={ data }
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render using null and undefined data", () => {
    const data: SourceCostsModel = {
      ...baseSourceCostsModel,
      sourceRenewHighValue: null,
      sourceRenewLikelyValue: undefined,
      sourceRenewLowValue: null
    };

    const wrapper = shallow(
      <RenewalEstimation
        data={ data }
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render using values equal to 0", () => {
    const data: SourceCostsModel = {
      ...baseSourceCostsModel,
      sourceRenewHighValue: 0,
      sourceRenewLikelyValue: 0,
      sourceRenewLowValue: 0
    };

    const wrapper = shallow(
      <RenewalEstimation
        data={ data }
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

});
