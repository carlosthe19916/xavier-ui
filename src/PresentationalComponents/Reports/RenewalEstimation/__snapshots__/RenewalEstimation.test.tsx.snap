// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenewalEstimation expect to render 1`] = `
<Fragment>
  <Component
    aria-label="Renewal estimation"
    borders={false}
    cells={
      Array [
        "For vSphere Enterprise Plus",
        Object {
          "props": Object {
            "className": "pf-u-text-align-right",
          },
          "title": "ELA Renewal Estimation",
        },
      ]
    }
    rows={
      Array [
        Array [
          "-- As high as...",
          "$1,000",
        ],
        Array [
          "-- Most likely...",
          "$2,000",
        ],
        Array [
          "-- As low as...",
          "$3,000",
        ],
      ]
    }
    variant="compact"
  >
    <TableHeader />
    <TableBody />
  </Component>
</Fragment>
`;

exports[`RenewalEstimation expect to render using null and undefined data 1`] = `
<Fragment>
  <Component
    aria-label="Renewal estimation"
    borders={false}
    cells={
      Array [
        "For vSphere Enterprise Plus",
        Object {
          "props": Object {
            "className": "pf-u-text-align-right",
          },
          "title": "ELA Renewal Estimation",
        },
      ]
    }
    rows={
      Array [
        Array [
          "-- As high as...",
          "Unknown",
        ],
        Array [
          "-- Most likely...",
          "Unknown",
        ],
        Array [
          "-- As low as...",
          "Unknown",
        ],
      ]
    }
    variant="compact"
  >
    <TableHeader />
    <TableBody />
  </Component>
</Fragment>
`;

exports[`RenewalEstimation expect to render using null and undefined data 2`] = `
<Fragment>
  <Component
    aria-label="Renewal estimation"
    borders={false}
    cells={
      Array [
        "For vSphere Enterprise Plus",
        Object {
          "props": Object {
            "className": "pf-u-text-align-right",
          },
          "title": "ELA Renewal Estimation",
        },
      ]
    }
    rows={
      Array [
        Array [
          "-- As high as...",
          "Unknown",
        ],
        Array [
          "-- Most likely...",
          "Unknown",
        ],
        Array [
          "-- As low as...",
          "Unknown",
        ],
      ]
    }
    variant="compact"
  >
    <TableHeader />
    <TableBody />
  </Component>
</Fragment>
`;

exports[`RenewalEstimation expect to render using values equal to 0 1`] = `
<Fragment>
  <Component
    aria-label="Renewal estimation"
    borders={false}
    cells={
      Array [
        "For vSphere Enterprise Plus",
        Object {
          "props": Object {
            "className": "pf-u-text-align-right",
          },
          "title": "ELA Renewal Estimation",
        },
      ]
    }
    rows={
      Array [
        Array [
          "-- As high as...",
          "$0",
        ],
        Array [
          "-- Most likely...",
          "$0",
        ],
        Array [
          "-- As low as...",
          "$0",
        ],
      ]
    }
    variant="compact"
  >
    <TableHeader />
    <TableBody />
  </Component>
</Fragment>
`;
